### ISSUE TITLE
Support datashares on Redshift

### ISSUE THREAD

**Issue by Tony-metabase** [metabase/metabase#47070]
**Is your feature request related to a problem? Please describe.**
Redshift database offers the functionality to define datashares, more info here:

https://docs.aws.amazon.com/redshift/latest/mgmt/query-editor-v2-datashare-using.html

You are able to access this datashare via SQL when using  the Native Query Editor and running `<shareddatabase>.<schema>.<object>`. So the jdbc driver knows about this datashare and how to fetch data. However when you go to create a question using a field filter, you can not see the schema/table from the data share to select the column. Also there are no Tables in the Admin -> Table Metadata.

Metabase simply doesn't know about this datashare since it doesn't run the proper queries to get the metadata for it.

**Describe the solution you'd like**
It would be nice to have a setting in the Database Settings, something like this:

<img width="1511" alt="image" src="https://github.com/user-attachments/assets/1c5a8322-d4ba-4008-aef2-696490f9cfe9">

In which if you enable Metabase will support Redshift datashares and run sync and scan queries accordingly.

**Describe alternatives you've considered**
Nothing just use the SQL but you end up very limited in functionality

**Comment by brunobergher** [comment]
I guess this effectively means including datashares as part of the entities enumerated during sync, right?

**Comment by Tony-metabase** [comment]
Yes correct. Right now during sync metabase doesn't know about//what datashares are ... Now we can either have this by default or give the option enable/disable it so metabase knows when and when not to access datashares

**Comment by fvianamo-ttl** [comment]
I am also interested! We rely a lot on data sharing and just starting to work with Metabase. While we do not have this native integration with Redshift external databases, if you create external schemas on Metabase's connected database your relations will appear on the next sync.
```
CREATE EXTERNAL SCHEMA <schema_name>
    FROM REDSHIFT DATABASE <external database created from datashare>
    SCHEMA <schema_name on datashare>;
```

**Comment by marshallm94** [comment]
Is there any update on this?

@fvianamo-ttl did you run your command in RedShift?

**Comment by fvianamo-ttl** [comment]
@marshallm94 yes!

**Comment by marshallm94** [comment]
@fvianamo-ttl did you run that in the _consumer_ redshift cluster/namespace, or the _producer_ redshift cluster?

**Comment by fvianamo-ttl** [comment]
@marshallm94 you run it at consumer side. I am assuming that you have already created a database on your consumer cluster, based on a datashare. Then, you have to mirror schemas that you introduced with your shared database by creating those proposed external schemas on the database you connect Metabase on.

**Comment by marshallm94** [comment]
@fvianamo-ttl worked great - thanks!

For anyone else that stumbles here from Google:

Assumptions:
- 1. A datashare has been created from the producer cluster/namespace to the consumer cluster/namespace.
- 2. A database has been created in the consumer cluster/namespace from the datashare.

- Steps:
  - 1. In the consumer cluster/namespace, run the following (making necessary substitutions):
    ```sql
    CREATE EXTERNAL SCHEMA {schema name for consumer cluster/namespace} FROM REDSHIFT DATABASE {external database created from datashare} SCHEMA {schema name in datashare};
    ```
  - 2. Grant usage on the external schema **to the user that is used to connect to the DB in Metabase**:
    ```sql
    GRANT USAGE ON SCHEMA {schema name for consumer cluster/namespace} TO GROUP {group that is used to connect to consumer cluster/namespace in Metabase};
    ```

This worked for me on Metabase Version: v0.46.6.4 using RedShift Serverless
